modules/core/src/test/java/org/apache/ignite/internal/processors/security/impl/TestCertificateSecurityProcessor.java

```diff
@@ -70,14 +73,18 @@ public TestCertificateSecurityProcessor(GridKernalContext ctx, Collection<TestSe
 
     /** {@inheritDoc} */
     @Override public SecurityContext authenticateNode(ClusterNode node, SecurityCredentials cred) {
-        return new TestSecurityContext(
+        SecurityContext res = new TestSecurityContext(
             new TestSecuritySubject()
                 .setType(REMOTE_NODE)
                 .setId(node.id())
                 .setAddr(new InetSocketAddress(F.first(node.addresses()), 0))
                 .setLogin("")
                 .setPerms(ALLOW_ALL)
         );
+
+        SECURITY_CONTEXTS.put(res.subject().id(), res);
```

Comment:
Why `SECURITY_CONTEXTS` made static?
AFAICS it used in a non-static context only.